qa/L0_model_config/autofill_noplatform_success/onnx/cpu_instance/expected

```diff
@@ -32,4 +32,5 @@ optimization {
     enable: true
   }
 }
-backend: "onnxruntime"
\ No newline at end of file
+backend: "onnxruntime"
+runtime: ""
```

Comment:
should auto-complete fill in this field with correct value as well (i.e., this would be `/opt/tritonserver/backends/onnxruntime/libtriton_onnxruntime.so`)?

Reply:
That was my original thought, but the issue is how does auto-complete determine if the model uses C++ or Python runtime? A simple solution is to have backend specific auto-complete implementation. For example, if it the model uses PyTorch backend and a `*.pt` model file is provided without any `*.py` file, then use C++. If a `*.py` file is provided, then use Python. See [this commit](https://github.com/triton-inference-server/core/pull/285/commits/2e5b50963f4cd41b440a4359a1d038256f25179f). From previous discussions with @nnshah1 and @rmccorm4, we want to stop adding more backend specific logic into auto-complete and opt for a generic implementation, which Python runtime falls into the category of [custom backend](https://github.com/triton-inference-server/core/pull/285/files#diff-f947d03afcf01249667bedc5b955404da083564c22809b6666ba923356f7c829R1179) on auto-complete. Backends can implement either C++ or Python runtime (or both), we do not want to auto-complete into Python runtime on a C++ only backend, which requires checking the backend installation directory. The current structure limits auto-complete to only see model directory, but not backend directory, because the latter is resolved only when the model is loaded, which happens after auto-complete. A easier approach is to resolve runtime, if not already filled, when the model is loaded, which has both the model and backend directory information.

Going back to the original question, auto-complete does not touch the runtime field, it is determined/filled when the model is loaded, if it is not already filled.